src/shared/services/commandInterpreterHelper.js

```diff
@@ -384,18 +384,22 @@ const availableCommands = [
     exec: (action, cmdchar, put, store) => {
       const state = store.getState()
 
-      // Since we now also handle queries with the :auto prefix, we need to strip that
-      // and attach to the actions object
-      const query = action.cmd.replace(
-        getCmdChar(state) + autoCommitTxCommand,
-        ''
-      )
-      action.query = query.trim()
+      // Since we now also handle queries with the :auto prefix,
+      // we need to strip that and attach to the actions object
+      // note that comment removal is crude and will remove
+      // false positives inside strings. but it's good enough
+      // for our use case here
+      const query = action.cmd.trim()
+
+      const autoPrefix = getCmdChar(state) + autoCommitTxCommand
+      const isAutocommit = query
+        .replace(/\/\*(.|\n)*?\*\//g, '') // mutliline comment
+        .replace(/\/\/[^\n]*\n/g, '') // singleline comment
+        .trim()
+        .startsWith(autoPrefix)
 
-      // We need to find out if this is an auto-committing tx or not
-      // i.e. Did we strip something off above here?
-      const autoCommit = action.query.length < action.cmd.trim().length
-      action.autoCommit = autoCommit
+      action.autoCommit = isAutocommit
+      action.query = isAutocommit ? query.replace(autoPrefix, '').trim() : query
```

Comment:
I think replacing first instance of a text will fail if a comment has the text being replaced, e.g.
```
// browser should remove :auto in the query below this comment
:auto CREATE (n:Test {val: 'yo:auto'}) RETURN *
```

Reply:
Right. Thinking of how this is best solved without a proper parse.. I'll send an update, do tell what you think

Reply:
Hmm... none of these handle the case of `:auto return 1; :auto return 1;`. Is that statement legal with multiline enabled?

Reply:
Not sure what the best way forward (barring a proper parse) is. This query for example bypasses the easy-ish solutions I can think of.. `/* :auto */ :auto RETURN 1;`

Reply:
The issue I noted is already an issue in production, this fix does not cause any extra issue